services/document-manager/backend/app/docman/utils/document_upload_helper.py

```diff
@@ -13,15 +13,41 @@
 from flask import request, current_app, make_response, jsonify
 from werkzeug.exceptions import BadRequest, NotFound, RequestEntityTooLarge, InternalServerError, BadGateway
 
+from datetime import datetime
+
+from werkzeug.exceptions import BadRequest, BadGateway, InternalServerError
+from flask import request, current_app
+
+from app.extensions import db
+from app.config import Config
+from app.services.object_store_storage_service import ObjectStoreStorageService
+from app.docman.models.document import Document
+from app.docman.models.document_version import DocumentVersion
+from app.services.object_store_storage_service import ObjectStoreStorageService
+from werkzeug.utils import secure_filename
 CACHE_TIMEOUT = TIMEOUT_24_HOURS
 
+def handle_status_and_update_doc(status, doc_guid):
+    doc = Document.find_by_document_guid(doc_guid)
+    db.session.rollback()
+    doc.status = str(status)
+    db.session.add(doc)
+    db.session.commit()
+
 
 class DocumentUploadHelper:
 
     @classmethod
     def initiate_document_upload(cls, document_guid, file_path, folder, file_size, version_guid=None):
+        folder = secure_filename(folder) if folder else folder
```

Comment:
Is this what's causing the the code scanning check failure below? The else path particularly? I think that's the first time I've seen that check raise an issue!

Reply:
Indeed. The `folder` parameter left without `secure_filename` would cause the check to fail - as it tries to create that folder on the filesystem (only for local development). Updated the fallback to `None` - check passes now.

Reply:
That folder logic has been there for ages, so I guess the codescanning was enabled after this was last changed 😓